crates/sui-framework/sources/coin.move

```diff
@@ -224,6 +224,17 @@ module sui::coin {
         )
     }
 
+    /// Split coin `self` into `n` coins with equal balances, or as close as
+    /// possible, if the balance is not evenly divisible by `n`.
+    public entry fun split_n<T>(self: &mut Coin<T>, n: u64, ctx: &mut TxContext) {
+        let i = 0;
+        let split_amount = balance::value(&self.balance) / n;
+        while (i < n - 1) {
+            split(self, split_amount, ctx);
+            i = i + 1;
+        };
+    }
```

Comment:
I think a useful, related function would be something like

```
public fun split_n<T>(self: &mut Coin<T>, n: u64, ctx: &mut TxContext): vector<Coin<T>>
```

, since this version allows the caller to continue using the split pieces instead of immediately transferring them.

Might make sense to have both API's and implement one in terms of the other.

Reply:
PS open to suggestions for what to name the new variant :)

Reply:
> PS open to suggestions for what to name the new variant :)

I'd say that `split_n_return`'s signature already implies that the split coins are returned so I'd change it to simply `split_n` and have the other one called `split_n_transfer` as it's unclear from signature what happens inside beyond splitting.

Reply:
I thought about that but was worried it would be confusing because `split` and `split_vec` both transfer without having `transfer` in their name. So if this one has a `_transfer` suffix then it might make it seem like the others *don't* transfer? What do you think?

Reply:
Good point. It's probably more consistent to leave it as is in this case. It bothers me a bit that `split_n_return` is somewhat redundant, but I am not super religious about these things :-)

Reply:
No argument here that the name isn't ideal! Maybe `split_n_into_vec`? That gets a bit wordy, but at least it's clear.

Reply:
> split_n_into_vec

Perhaps save a syllable and make it `split_n_to_vec` with the "in" implied?

Reply:
I like it, done!